File: java/lab5/src/baseabstractions/Builder.java
package src.baseabstractions;

/**
 * Interface for the builder pattern, providing methods to construct objects.
 *
 * @param <T> the type of object that this builder creates
 */
public interface Builder<T> {

    /**
     * Resets the builder to its initial state.
     * Clears all previously set fields and prepares for building a new object.
     */
    void reset();

    /**
     * Creates and returns the built object.
     *
     * @return the constructed object of type T
     */
    T create();
}

File: java/lab8/gradle/client/src/main/kotlin/abstractions/Builder.kt
package abstractions

/**
 * Generic builder pattern interface.
 *
 * @param T the type of object to build
 */
interface Builder<T> {

    /**
     * Resets the builder to its initial state so a new object can be constructed.
     */
    fun reset()

    /**
     * Constructs and returns the built object.
     *
     * @return the newly created object
     */
    fun build(): T
}
